plugins/generator-1.16.5/forge-1.16.5/templates/item.java.ftl

```diff
@@ -329,6 +348,38 @@ import net.minecraft.entity.ai.attributes.Attributes;
         }
         </#if>
 
+        <#if (hasProcedure(data.onFinishUsingItem) || (data.eatResultItem?? && !data.eatResultItem.isEmpty()))>

```

Comment:
If one chooses food, defines eatResultItem and then unselects food, this part will still generate, but should not, same for 1.18.2